packages/runtime/src/runtime.ts

```diff
@@ -727,6 +719,7 @@ export class Runtime {
 	): TxReceipt {
 		this.addCtxAppUpdateTx(senderAddr, appID, payFlags, flags);
 		this.ctx.debugStack = debugStack;
+		this.ctx.budget = MAX_APP_PROGRAM_COST;
```

Comment:
Same as above. I think instead of doing this in each function, we should have a global function (eg. `computeTxBudget(..)`), which computes the `this.ctx.budget` depending on the execution mode and type of transaction. It will reduce the complexity in each function

Reply:
Good idea! 1000 👍🏻